### ISSUE TITLE
(🐞) false error: `undefined-loop-variable` when for loop has terminating else branch

### ISSUE THREAD

**Issue by KotlinIsland** [PyCQA/pylint#5971]
### Bug description
```py
def foo(l):
    for i in l:
        break
    else:
        return
    print(i)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:6:10: W0631: Using possibly undefined loop variable 'i' (undefined-loop-variable)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.13.0
astroid 2.11.1
Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
```

**Comment by DanielNoord** [comment]
Related are #5608, #5146.

@jacobtylerwalls Please help me remember, what was our conclusion about this message. Were you going to take a look at the larger problem with this message and these patterns for `2.14`?
